abci/types/types.proto

```diff
@@ -311,6 +311,22 @@ message ValidatorUpdate {
 message VoteInfo {
   Validator validator = 1 [(gogoproto.nullable)=false];
   bool signed_last_block = 2;
+  Vote full_vote = 3;
+}
+
+// Validator's vote for a block
+message Vote {
+    int32 type = 1;
```

Comment:
we are working on restructuring the data in the vote here: https://github.com/tendermint/tendermint/pull/3958, you can comment with any further ideas.